topydo/lib/Filter.py

```diff
@@ -214,3 +214,35 @@ def match(self, p_todo):
 
         return False
 
+PRIORITY_MATCH = r"\((?P<operator><=?|>=?|=|!)?(?P<priority>[A-Z]{1})\)"
+
+class PriorityFilter(Filter):
+    def __init__(self, p_expression):
+        super(PriorityFilter, self).__init__()
+
+        self.expression = p_expression
+
+        match = re.match(PRIORITY_MATCH, self.expression)
+        if match:
+            self.operator = match.group('operator') or '='
+            self.priority = match.group('priority')
+
+    def match(self, p_todo):
+        operand1 = p_todo.priority()
+        operand2 = self.priority
+
+        if self.operator == '<':
```

Comment:
This piece is identical to the code in `OrdinalTagFilter`, perhaps this part can be reused in a superclass, e.g. `OrdinalFilter`, with subclasses `OrdinalTagFilter` and `OrdinalPriorityFilter` (or just `PriorityFilter` as you wrote). The subclasses take care of setting the operands, while the base class evaluates the operator and the operands.

Because I can think of a future extension to filter on creation date, which would use the same functionality, only a different attribute to extract from the todo.